arch/risc-v/src/common/riscv_saveusercontext.S

```diff
@@ -24,27 +24,42 @@
 
 #include <nuttx/config.h>
 
-#include <arch/syscall.h>
+#include "riscv_macros.S"
 
 /****************************************************************************
- * Public Functions
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Symbols
  ****************************************************************************/
 
 /****************************************************************************
  * Name: up_saveusercontext
  *
  * Description:
- *   Save the current thread context.  Full prototype is:
- *
- *   int  up_saveusercontext(void *saveregs);
- *
- * Returned Value:
- *   0: Normal return
- *   1: Context switch return
+ *   Save the current thread context
  *
  ****************************************************************************/
 
-int up_saveusercontext(void *saveregs)
-{
-  return sys_call1(SYS_save_context, (uintptr_t)saveregs);
-}
+  .section .text
+  .global up_saveusercontext
+  .align  8
+
+up_saveusercontext:
+
+  save_ctx   a0
+
+  csrr       a1, CSR_STATUS
```

Comment:
Maybe a dumb question but doesn't this destroy a1 in the caller's context ? Should you not REGLOAD a1, REG_A1(a0) before returning (jr) ? (Sorry if this is a duplicate comment)

Reply:
Hi, from the risc-v calling convention, we can see a0-a1 is caller registers, so no need to do REGLOAD.

Reply:
Ok so even if the function takes only 1 parameter the caller must save both a0 and a1. Thanks.